src/core/network.js

```diff
@@ -414,9 +415,27 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition =
+      fullRequestXhr.getResponseHeader('Content-Disposition');
+    this._fileName = this._extractNameFrom(contentDisposition);
+
     this._headersReceivedCapability.resolve();
   },
 
+  _extractNameFrom:
+      function PDFNetworkStreamFullRequestReader_extractNameFrom(cDisposition) {
+    if (cDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(cDisposition);
```

Comment:
That's exactly what I was assuming @stemwinder . 

@Snuffleupagus Adding this validation in the method that extracts the file name from the headers in practice means that it will be ignored (it will return nil since it don't matches the input string). If you try to load something that is not a pdf it will fail elsewhere because of the file content.

If we want to add  a validation  on the file name we should go one step forward and stop the loading process if the received file name do not seems to be a valid pdf.

Anyway, if all this discussion is about adding a '\.pdf' in the regexp and writting a couple of tests, I'll do. It's not a big challenge at all.